types/abi/src/recode_rlp.rs

```diff
@@ -207,6 +213,47 @@ impl Abi {
     }
 }
 
+pub fn rlp_encode(input: Vec<u8>) -> Vec<u8> {
```

Comment:
Can this be zero-copy?

Take a mutable ref as input and slice the bytes without the allocation at L219